test/OrchardCore.Tests/Apis/ContentManagement/DeploymentPlans/ContentStepLuceneQueryTests.cs

```diff
@@ -50,5 +51,88 @@ public async Task ShouldUpdateLuceneIndexesOnImport()
+                jItem[nameof(ContentItem.DisplayText)] = "new version";
+            });
+            const string secondcontentitemversionid = "secondcontentitemversionid";
+            var data = recipe["steps"][0]["Data"].AsArray();
+            var secondContentItem = JObject.FromObject(context.OriginalBlogPost);
+            secondContentItem[nameof(ContentItem.ContentItemVersionId)] = secondcontentitemversionid;
+            secondContentItem[nameof(ContentItem.DisplayText)] = "second content item version display text";
+            secondContentItem[nameof(AutoroutePart)][nameof(AutoroutePart.Path)] = "new-path";
+            data.Add(secondContentItem);
+
+            Assert.Equal(2, data.Count);
+            Assert.Single(data.Select(x => x.SelectNode(nameof(ContentItem.ContentItemId)).ToString()).Distinct());
+
+            // validate different contentItem versionId
+            var versionIds = data.Select(x => x.SelectNode(nameof(ContentItem.ContentItemVersionId)).ToString()).Distinct();
+            Assert.Equal(2, versionIds.Count());
+
+            await context.PostRecipeAsync(recipe);
+
+            // Search indexes are no longer updated in a deferred task at the end of a shell scope
+            // but in a background job after the http request, so they are not already up to date.
+
+            await context.UsingTenantScopeAsync(async scope =>
+            {
+                var luceneQuerySource = scope.ServiceProvider.GetRequiredService<LuceneQuerySource>();
+                var luceneQuery = new LuceneQuery
+                {
+                    Index = "Search",
+                    ReturnContentItems = false,
+                    Template = @"
+                            {
+                                ""query"": {
+                                    ""term"": { ""Content.ContentItem.ContentType"": ""BlogPost"" }
+                                },
+                                ""filter"": [
+                                    {
+                                        ""term"": { ""Content.ContentItem.Published"":""true"" }
+                                    },
+                                    {
+                                        ""term"": { ""Content.ContentItem.ContentItemId"":""{{parameters.contentItemId}}"" }
+                                    }
+                                ]
+                            }",
+                };
+                var queryParameters = new Dictionary<string, object>
+                {
+                    ["contentItemId"] = context.OriginalBlogPost.ContentItemId
+                };
+                var timeoutTask = Task.Delay(5_000);
+                while (true)
+                {
+                    await Task.Delay(1_000);
+                    // Test
+                    var result = (LuceneQueryResults)await luceneQuerySource.ExecuteQueryAsync(luceneQuery, queryParameters);
```

Comment:
Using the `LuceneQuerySource` directly, avoid  to call `ContentManagement.GetAsync`